clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -503,6 +508,20 @@
     private static final long NO_CURRENT_THREAD = -1L;
     private static final AtomicInteger CONSUMER_CLIENT_ID_SEQUENCE = new AtomicInteger(1);
     private static final String JMX_PREFIX = "kafka.consumer";
+    /**
+     * APIs used by KafkaConsumer
+     */
+    private static final List<Short> USED_API_KEYS = Arrays.asList(
+            ApiKeys.METADATA.id,
+            ApiKeys.FETCH.id,
+            ApiKeys.GROUP_COORDINATOR.id,
+            ApiKeys.HEARTBEAT.id,
+            ApiKeys.JOIN_GROUP.id,
+            ApiKeys.LEAVE_GROUP.id,
+            ApiKeys.LIST_OFFSETS.id,
+            ApiKeys.OFFSET_COMMIT.id,
+            ApiKeys.OFFSET_FETCH.id,
+            ApiKeys.SYNC_GROUP.id);
```

Comment:
Added system test against trunk and older broker versions. The interesting tests like testing against unsupported broker versions, will probably have to wait till we have such versions.

The list of APIs used by client is similar to the practice of specifying used errors for a request. As you said, it will be nice to have this checked dynamically at test time, however nothing comes to my mind with which we can do that. Will be happy to accommodate any suggestion on this.